model/config.go

```diff
@@ -1831,6 +1831,7 @@ type TeamSettings struct {
 	EnableOpenServer                                          *bool    `access:"authentication"`
 	EnableUserDeactivation                                    *bool    `access:"experimental"`
 	RestrictCreationToDomains                                 *string  `access:"authentication"`
+	EnableCustomUserStatuses                                  *bool    `access:"site"`
```

Comment:
Each new feature setting should be behind a Feature Flag, please following instructions here: https://developers.mattermost.com/contribute/server/feature-flags/

Reply:
@reflog The doc mentions that each feature flag should be off by default.
But, we need to make sure the setting is enabled for all users.

Any advice there?

Reply:
@chetanyakan Feature flags override the Config defaults. Currently, any new feature that is added to Mattermost is first 'hidden' behind a feature flag, rolled out gradually, tested, and then the feature flag is removed completely, leaving the defaults that you've set in the Config to take effect

Reply:
@reflog I can set the feature flag to turn the custom status feature off.

Is there a way I can set the feature flag to enable it locally and on the test server?

Reply:
@chetanyakan - you take https://github.com/mattermost/mattermost-server/pull/16598 as an example of how to add a feature flag